### DISCUSSION TITLE
Error Tests Parte 2 "IN-X-RANGE" {A} {B}

### DISCUSSION THREAD

**Discussion by tsunayoshi21** [IIC2133-PUC/2024-1#74] [Tarea 1] [answered]
Hola! Creo que hay un error en los test de la parte 2 en la parte de  "IN-X-RANGE" {A} {B}.

Según enunciado, se debe entregar todas las cabezas tal que A<= x <= B. Y creo que eso no se refleja en los test de esa parte.
Por ejemplo en el test "4-x_range-sample-ayudantes"
Se tiene el siguiente input:

20
SAVE 7 PabloNavarro 1997 2 17 12
SAVE 21 JavieraLarraín 2001 2 9 22
SAVE 19 EstefaniaPakarati 2005 1 17 26
SAVE 26 HeinzBusch 1998 2 10 7
SAVE 15 PaulaGrune 1996 2 13 30
SAVE 13 AndrésMoraga 1998 2 5 30
SAVE 9 Anitamarti 2005 2 30 3
SAVE 18 AlexanderInfanta 2002 2 31 21
SAVE 12 AgustínGutiérrezCastro 2000 1 31 29
SAVE 28 FranciscaDeLaBarra 2002 1 9 2
SAVE 0 CarlosParedes 2001 2 24 5
SAVE 3 BenjaVicente 1996 1 31 6
SAVE 20 LucíaFardella 1999 2 4 7
SAVE 11 SebastianAchondo 2001 2 24 23
SAVE 1 CataAlamos 1998 1 6 5
SAVE 16 IsabellaCherubini 2004 1 0 12
SAVE 4 BernarditaJesúsNazar 2000 1 29 11
SAVE 22 VicenteThomas 1995 2 10 15
SAVE 8 BorjaMárquezdelaPlata 2003 2 4 24
SAVE 23 FelipeEspinoza 2000 2 23 29
3
IN-X-RANGE 0 1
IN-X-RANGE 12 15
IN-X-RANGE 5 6

Y este debería ser el output correcto según el test:

IN-X-RANGE 0 1: 1
16 IsabellaCherubini
IN-X-RANGE 12 15: 1
15 PaulaGrune
IN-X-RANGE 5 6: 1
13 AndrésMoraga

Sin embargo en la parte de "IN-X-RANGE 5 6" falta 
1 CataAlamos
Ya que esta cabeza fue registrada como 
SAVE 1 CataAlamos 1998 1 6 5

Por lo que 5 <= x <=6, sin embargo no se incluye en el output correcto.

Esto también ocurre en los otros tests de esa parte pero son muy largos para citarlos.
Al mismo tiempo, en los otros test de esa parte ocurre que no están ordenados por menor valor de x como dice el enunciado. En específico el test "4-x_range-futurama" Tiene como output:

IN-X-RANGE 41 47: 13
51 ChrisTravers
47 Beelzebot
35 Hookerbot

Y estas cabezas fueron inicializadas como:

SAVE 51 ChrisTravers 2997 4 42 30
SAVE 47 Beelzebot 2993 1 44 45
SAVE 35 Hookerbot 2999 3 41 39

Por lo tanto, en los test no se respeta ni el orden por menor valor de x, ni estan todas las cabezas en el rango estipulado. Me gustaría saber si estoy en lo correcto o no entendí bien el problema.

Graciass.

**Comment by benjavicente** [comment, marked as answer]
Hola! Es $A<= x < B$, aunque efectivamente el orden estaba malo.

Subimos unos nuevos con el arreglo ahora, verifica si están ok y nos avisas plz.

**Reply by tsunayoshi21** [reply]
Hola! Si, ahora con los cambios me funcionó todo bien. Graciass
